Sources/SwiftSyntax/SyntaxBaseNodes.swift.gyb

```diff
@@ -59,12 +59,6 @@ public struct ${node.name}: ${node.name}Protocol, SyntaxHashable {
     self.init(syntax._syntaxNode.data)
   }
 
-  /// Create a `${node.name}` node from a specialized optional syntax node.
-  public init?<S: ${node.name}Protocol>(_ syntax: S?) {
-    guard let syntax = syntax else { return nil }
-    self.init(syntax)
```

Comment:
I'm not sure we should delete this.
This exists for delegating to specialized non-nil `init(_:)`. For example,
```
func test(declOpt: FuncDeclSyntax?) -> DeclSyntax? {
  return DeclSyntax(declOpt)
}
```
should call `DeclSyntax.init<S: DeclSyntaxProtocol>(_: S)`, but not `.init?<S: SyntaxProtocol>(_: S)`.
I don't think `SyntaxProtocol.init?<S: SyntaxProtocol>(_: S)` would do it.

Reply:
Ah, I suppose we'd be skipping the fast path in this case yeah. Fair enough, I'll add it back in.